src/ExpenseTracker.Core/Manager/TransactionManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Threading.Tasks;
+using System.Transactions;
+using ExpenseTracker.Core.Dto.Transaction;
+using ExpenseTracker.Core.FileManager.Interface;
+using ExpenseTracker.Core.Manager.Interface;
+using ExpenseTracker.Core.Services.Interface;
+using Microsoft.AspNetCore.Http;
+
+namespace ExpenseTracker.Core.Manager
+{
+    public class TransactionManager : ITransactionManager
+    {
+        private readonly ITransactionService _transactionService;
+        private readonly IFileManager _fileManager;
+
+        public TransactionManager(ITransactionService transactionService, IFileManager fileManager)
+        {
+            _transactionService = transactionService;
+            _fileManager = fileManager;
+        }
+
+        public async Task RecordTransaction(IFormFile? file, TransactionCreateDto dto)
+        {
+            using var tsc = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            if (file != null)
```

Comment:
@BirajMainali Maybe we can simplify this branching logic? We don't need to perform transaction create in two places if we just move it outside of the branching. This also removes the if/else. We only need an `if` for file purpose